modules/connectivity/ParticipantConnectionStatus.js

```diff
@@ -271,70 +279,87 @@ export default class ParticipantConnectionStatus {
+        }
+
+        const hasAnyVideoRTCMuted = participant.hasAnyVideoTrackWebRTCMuted();
+        var rtcMutedTimestamp
+            = this.rtcMutedTimestamp[participant.getId()];
+
+        return hasAnyVideoRTCMuted
+            && typeof rtcMutedTimestamp === 'number'
+            && (Date.now() - rtcMutedTimestamp) >= this.rtcMuteTimeout;
+    }
+
+    /**
+     * Figures out (and updates) the current connectivity status for
+     * the participant identified by the given id.
+     *
+     * @param {string} id the participant's id (MUC nickname or Colibri endpoint
+     * ID).
+     */
+    figureOutConnectionStatus(id) {
+        var participant = this.conference.getParticipantById(id);
         if (!participant) {
             // Probably the participant is no longer in the conference
             // (at the time of writing this code, participant is
             // detached from the conference and TRACK_REMOVED events are
             // fired),
             // so we don't care, but let's print the warning for
             // debugging purpose
-            logger.warn(
-                'maybeRestoreCachedStatus - ' +
-                'no participant for endpoint: ' + endpointId);
+            logger.warn('figure out conn status - no participant for: ' + id);
             return;
         }
 
-        const isConnectionActive = participant.isConnectionActive();
-        const hasAnyVideoRTCMuted = participant.hasAnyVideoTrackWebRTCMuted();
-        let isConnActiveByJvb = this.rtcConnStatusCache[endpointId];
+        const isVideoMuted = participant.isVideoMuted();
+        const isVideoTrackFrozen = this.isVideoTrackFrozen(participant);
+        let isConnActiveByJvb = this.connStatusFromJvb[id];
 
         // If no status was received from the JVB it means that it's active
         // (the bridge does not send notification unless there is a problem).
         if (typeof isConnActiveByJvb !== 'boolean') {
-            logger.debug("Assuming connection active by JVB - no notification");
+            logger.debug('Assuming connection active by JVB - no notification');
             isConnActiveByJvb = true;
         }
 
-        logger.debug(
-            "Remote track removed, is active: " + isConnectionActive
-            + " is active(jvb):" + isConnActiveByJvb
-            + " video RTC muted:" + hasAnyVideoRTCMuted);
-
-        if (!isConnectionActive && isConnActiveByJvb && !hasAnyVideoRTCMuted) {
-            // FIXME adjust the log level or remove the message completely once
-            // the feature gets mature enough.
-            logger.info(
-                "Remote track removed for disconnected" +
-                " participant, when the status according to" +
-                " the JVB is connected. Adjusting to the JVB value.");
-            this._changeConnectionStatus(endpointId, isConnActiveByJvb);
+        var isConnectionActive = true;
+        if (!isVideoMuted && (isVideoTrackFrozen || !isConnActiveByJvb)) {
```

Comment:
Probably it is, but I can't easily confirm that this one line statement is the same thing, so I find it more readable in separate "ifs". That's because there are 2 major cases when connection is considered "not active" and writing two separately allowed me to comment on each one of them. Maybe the comments could be improved though...